icu4c/source/test/cintltst/ccaltst.c

```diff
@@ -100,6 +100,7 @@ static const UCalGetTypeTest ucalGetTypeTests[] = {
     { "fr_CH",                   UCAL_DEFAULT,   "gregorian" },
     { "fr_SA",                   UCAL_DEFAULT,   "islamic-umalqura" },
     { "fr_CH@rg=sazzzz",         UCAL_DEFAULT,   "islamic-umalqura" },
+    { "fr_CH@rg=sawxyz",         UCAL_DEFAULT,   "islamic-umalqura" }, /* ignore bogus subdivision tag (ICU-22364) */
```

Comment:
sawxyz is not listed in https://github.com/unicode-org/cldr/blob/main/common/supplemental/subdivisions.xml

how about 
```
{ "fr_CH@rg=sa14",         UCAL_DEFAULT,   "islamic-umalqura" },
```
instead and also try
rg as "twcyi" (listed in https://github.com/unicode-org/cldr/blob/main/common/supplemental/subdivisions.xml )
rg as "ugw"

Reply:
Okay, I'll change this and add the others you're suggesting.